src/ecwrapper.cpp

```diff
@@ -177,8 +179,14 @@ bool CECKey::Recover(const uint256 &hash, const unsigned char *p64, int rec)
     if (rec<0 || rec>=3)
         return false;
     ECDSA_SIG *sig = ECDSA_SIG_new();
-    BN_bin2bn(&p64[0],  32, sig->r);
-    BN_bin2bn(&p64[32], 32, sig->s);
+    BIGNUM *sig_r, *sig_s;
```

Comment:
Set these to nullptr. Otherwise, it is possible that the first ``BN_bin2bn`` fails and ``sig_s`` is accessed when uninitialized, which is undefined behaviour. ``BN_free(NULL)`` is [defined to be a no-op](https://www.openssl.org/docs/manmaster/crypto/BN_free.html).

(This is not a security bug in current Bitcoin because they already removed the use of OpenSSL.)